examples/machinedeployment/machinedeployment.yaml

```diff
@@ -3,20 +3,19 @@ kind: MachineDeployment
 metadata:
   name: ${CLUSTER_NAME}-md-0
   labels:
-    cluster.x-k8s.io/cluster-name: ${CLUSTER_NAME}
     nodepool: nodepool-0
 spec:
+  clusterName: "${CLUSTER_NAME}"
   replicas: 1
   selector:
     matchLabels:
-      cluster.x-k8s.io/cluster-name: ${CLUSTER_NAME}
```

Comment:
we probably want to keep this in matchlabels to avoid accidentally matching a machines that belong to a different cluster if someone re-uses values for `nodepool` label for example.

Either that or we'll need to start appending the cluster label to the provided matchLabels implicitly in the machinedeployment/machineset controllers.

Reply:
That's a good point, what do you think of doing that automatically in CAPI?

Reply:
No objections, but it will need to likely be reflected in the generated `Status.Selector` fields as well